adapters/coap/src/main/java/org/eclipse/hono/adapter/coap/CoapContext.java

```diff
@@ -398,6 +400,19 @@ public void respondWithCode(final ResponseCode responseCode, final String descri
         respond(response);
     }
 
+    /**
+     * Checks whether the request includes either the time option or the hono-time parameter which
+     * indicates that the response should include a time option with the server time.
+     *
+     * @return <code>true</code> if response should include the time option containing the server time.
+     */
+    private boolean shouldResponseIncludeTimeOption() {
+        return Optional.ofNullable(exchange.getRequestOptions())
+                .map(opts -> opts.hasOption(TimeOption.NUMBER))
+                .orElse(false)
+                || exchange.getQueryParameter(TimeOption.QUERY_PARAMETER_NAME) != null;
```

Comment:
Could this be done like this instead:

```java
return Optional.ofNullable(exchange.getRequestOptions())
                .map(opts -> opts.hasOption(TimeOption.NUMBER))
                .orElse(exchange.getQueryParameter(TimeOption.QUERY_PARAMETER_NAME) != null);
```
Is that in any way clearer? Or maybe just worse?